### ISSUE TITLE
`brew doctor` is confused about files in /usr/local/lib after installing via rustup

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust#25130]
I know rust technically doesn't interact with homebrew, but `brew doctor` returns warnings about unknown files (which are used by Rust) existing in `/usr/local/lib`. Removing these unknown files causes Rust to stop working. *Ideally*, something could be done to prevent those files from showing up via `brew doctor` to prevent new users from deleting them.

Steps to reproduce

```
~ $ sudo sh /usr/local/lib/rustlib/uninstall.sh
install: uninstalling component 'rustc'
install: uninstalling component 'cargo'
install: uninstalling component 'rust-docs'

    Rust is uninstalled.
```
```
~ $ brew doctor
Your system is ready to brew.
```
```
~ $ curl -s https://static.rust-lang.org/rustup.sh | sh -s

Welcome to Rust.

This script will download the Rust compiler and its package manager, Cargo, and
install them to /usr/local. You may install elsewhere by running this script
with the --prefix=<path> option.

The installer will run under 'sudo' and may ask you for your password. If you do
not want the script to run 'sudo' then pass it the --disable-sudo flag.

You may uninstall later by running /usr/local/lib/rustlib/uninstall.sh,
or by running this script again with the --uninstall flag.

Continue? (y/N) y

rustup: gpg available. signatures will be verified
rustup: downloading manifest for 'beta'
rustup: downloading toolchain for 'beta'
######################################################################## 100.0%
gpg: assuming signed data in '/Users/coreyf/.rustup/dl/5b96c9bf1ab9ef140fa0/rust-1.0.0-beta.4-x86_64-apple-darwin.tar.gz'
gpg: Signature made Fri May  1 03:58:08 2015 EDT using RSA key ID 7B3B09DC
gpg: Good signature from "Rust Language (Tag and Release Signing Key) <rust-key@rust-lang.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 108F 6620 5EAE B0AA A8DD  5E1C 85AB 96E6 FA1B E5FE
     Subkey fingerprint: C134 66B7 E169 A085 1886  3216 5CB4 A934 7B3B 09DC
rustup: extracting installer
rustup: installing toolchain for 'beta'
install: creating uninstall script at /usr/local/lib/rustlib/uninstall.sh
install: installing component 'rustc'
install: installing component 'cargo'
install: installing component 'rust-docs'

    Rust is ready to roll.
```
```
~ $ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
    /usr/local/lib/libarena-4e7c5e5c.dylib
    /usr/local/lib/libflate-4e7c5e5c.dylib
    /usr/local/lib/libfmt_macros-4e7c5e5c.dylib
    /usr/local/lib/libgetopts-4e7c5e5c.dylib
    /usr/local/lib/libgraphviz-4e7c5e5c.dylib
    /usr/local/lib/liblog-4e7c5e5c.dylib
    /usr/local/lib/librbml-4e7c5e5c.dylib
    /usr/local/lib/librustc-4e7c5e5c.dylib
    /usr/local/lib/librustc_back-4e7c5e5c.dylib
    /usr/local/lib/librustc_borrowck-4e7c5e5c.dylib
    /usr/local/lib/librustc_driver-4e7c5e5c.dylib
    /usr/local/lib/librustc_lint-4e7c5e5c.dylib
    /usr/local/lib/librustc_llvm-4e7c5e5c.dylib
    /usr/local/lib/librustc_privacy-4e7c5e5c.dylib
    /usr/local/lib/librustc_resolve-4e7c5e5c.dylib
    /usr/local/lib/librustc_trans-4e7c5e5c.dylib
    /usr/local/lib/librustc_typeck-4e7c5e5c.dylib
    /usr/local/lib/librustdoc-4e7c5e5c.dylib
    /usr/local/lib/libserialize-4e7c5e5c.dylib
    /usr/local/lib/libstd-4e7c5e5c.dylib
    /usr/local/lib/libsyntax-4e7c5e5c.dylib
    /usr/local/lib/libterm-4e7c5e5c.dylib
    /usr/local/lib/libtest-4e7c5e5c.dylib
```

**Comment by kballard** [comment]
This is Homebrew's problem. There's nothing Rust should attempt to do about it. All `brew doctor` is really saying there is "you've installed dylibs into Homebrew's prefix that Homebrew does not control", which is an issue that can affect any third-party software installation into `/usr/local` that includes dylibs and is not unique to Rust. You just don't see it very often because manually-installed 3rd-party software doesn't usually install dylibs into `/usr/local` (any dylibs it wants to use are usually expected to be provided by the user).

**Comment by frewsxcv** [comment]
> This is Homebrew's problem. 

Right. I understand this is technically not an issue with Rust. I would guess that the majority of Rust users on Macs have Homebrew installed, so preventing `brew doctor` from complaining about the Rust files from prevent a lot of confusion in the future. I'm not familiar with Homebrew nor the Rust build system, but I opened this hoping to hear from someone who is familiar with both how trivial this would be to address.

**Comment by reem** [comment]
For what it's worth, if you use multirust brew does not complain since multirust keeps the dylibs in a different place.

**Comment by kballard** [comment]
@frewsxcv I am familiar with both. The only solution (besides installing Rust using Homebrew, which does in fact already have a formula for it) would be to move the libraries somewhere else.

**Comment by frewsxcv** [comment]
Does anyone know if the `.pkg` file (linked from the Rust homepage) installs Rust to the same location as the `rustup` script?

**Comment by frewsxcv** [comment]
Sorry if this comes across as something I feel strongly about; it's not and I hope I don't give that impression. It was something I noticed from my personal experiences with Rust on OSX and didn't see this explicitly addressed anywhere. I would not be offended if this was closed due to it being a non-issue.

**Comment by steveklabnik** [comment]
Yes, while this is unfortunate, it's more of Homebrew's issue than ours.
